### DISCUSSION TITLE
Are all relations instances of either MaterialRelationshipType or ComparativeRelationshipType?

### DISCUSSION THREAD

**Discussion by ManuelSimoes** [nemo-ufes/gufo#33] [Q&A] [answered]
The [UFO documentation](https://ontouml.readthedocs.io/en/latest/relationships/intro.html) states that all relations can be divided in two broad categories, Material and Formal, but in the subsequent pages on different relations there's no explanation what this entails. 

Q1) Are all relations in gUFO either Material (MaterialRelationshipType) or Formal (ComparativeRelationshipType) as in UFO? 

Q2) Does this e.g. mean that isComponentOf is an instance of ComparativeRelationshipType? This seems counter-intuitive as mereology doesn't seem to be derived from IntrinsicAspects of the related entities, which is how ComparativeRelationshipType is defined. 

Q3) Does this e.g. mean that isComponentOf is an instance of MaterialRelationshipType? This seems counter-intuitive as mereology doesn't seem to be derived from ExtrinsicAspects of the related entities, which is how MaterialRelationshipType is defined.

**Comment by djradon** [comment]
Great question! I'm so glad there's a forum for discussing stuff like this!

For Q1, since there is a "gufo:RelationshipType" superclass, I would say that at least _technically_ you could have a relation that is neither Material or Formal. 

If I had to guess, I'd say isComponentOf might be considered a formal relationship. "Formal relations hold between two or more entities directly without any further intervening individual." isComponentOf seems to be direct.  "The relata of a material relation are mediated by individuals that are called relators. Relators are individuals with the power of connecting entities." I don't see what the extrinsic relator could be for isComponentOf, as you suggest. 

Also, it seems similar to some of the example formal relations: instantiation, inherence, association, and existential dependence.

**Comment by jpalmeida** [comment, marked as answer]
Concerning Q1, we have only included two subclasses of RelationshipType in gUFO, which are not exhaustive. ComparativeRelationshipType does not equate to formal relations in general, but specifically to "comparative formal relations" in [1].

Concerning Q2 and Q3: isComponentOf is a material relation. See [1] Section 7.4.

[1] https://research.utwente.nl/en/publications/ontological-foundations-for-structural-conceptual-models
